tools/depcheck/pkg/cmd/trace/pkg.go

```diff
@@ -0,0 +1,122 @@
+package trace
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/concrete"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+var (
+	baseRepoRegex = regexp.MustCompile("[a-zA-Z0-9]+\\.([a-z0-9])+\\/.+")
+)
+
+type Package struct {
+	Dir         string
+	ImportPath  string
+	Imports     []string
+	TestImports []string
+}
+
+type PackageList struct {
+	Packages []Package
+}
+
+func (p *PackageList) Add(pkg Package) {
+	p.Packages = append(p.Packages, pkg)
+}
+
+// BuildGraph receives a list of Go packages and constructs a dependency graph from it.
+// Any core library dependencies (fmt, strings, etc.) are not added to the graph.
+// Any packages whose import path is contained within a list of "excludes" are not added to the graph.
+// Returns a directed acyclic graph and a map of package import paths to node ids, or an error.
+func BuildGraph(packages *PackageList, excludes []string) (depgraph.MutableDirectedGraph, map[string]int, error) {
```

Comment:
I wouldn't expect this map to be separate.  Shouldn't it be part of the `mutableDirectedGraph`?